src/librustc_mir/const_eval.rs

```diff
@@ -527,6 +527,12 @@ pub fn error_to_const_error<'mir, 'tcx>(
     ConstEvalErr { error: error.kind, stacktrace, span: ecx.tcx.span }
 }
 
+pub fn undefined_behavior_error<'mir>() -> String {
+    "The rules on what exactly is undefined behavior aren't clear, \
+    so this check might be overzealous. Please open an issue on the rust compiler \
+    repository if you believe it should not be considered undefined behavior"
```

Comment:
The type signature of this function says `String` (an "owned" string), but the value here is a string literal, which has type `&str` (a reference to a "borrowed" string). I think `DiagnosticBuilder::note` [takes a `&str`](https://github.com/rust-lang/rust/blob/efb74579150a6ea0ad99a8568aa02a1aa23f6dd5/src/librustc_errors/diagnostic.rs#L189), so it's probably the type signature that you want to change.